manifest.json

```
@@ -1,144 +1,244 @@
-            },
-            "lifetime": "short",
-            "actions": [
-              {
-                "id": "TaskPaneRuntimeShow",
-                "type":"openPage",
-                "pinnable": false,
-                "view": "dashboard"
-              }
+        {
+          "name": "Document.ReadWrite.User",
+          "type": "Delegated"
+        }
+      ]
+    }
+  },
+  "validDomains": ["contoso.com"],
+  "extensions": [
+    {
+      "requirements": {
+        "scopes": ["mail", "workbook", "document", "presentation"]
+      },
+      "runtimes": [
+        {
+          "requirements": {
+            "capabilities": [
+              { "name": "Mailbox", "minVersion": "1.3" }
             ]
           },
-          {
-            "id": "CommandsRuntime",
-            "type": "general",
-            "code": {
-              "page": "https://localhost:3000/commands.html",
-              "script": "https://localhost:3000/commands.js"
-            },
-            "lifetime": "short",
-            "actions": [
-              {
-                "id": "action",
-                "type": "executeFunction"
-              }
-            ]
-          }
-        ],
-        "ribbons": [
-          {
-            "contexts": [
-              "mailRead"
-            ],
-            "tabs": [
-              {
-                "builtInTabId": "TabDefault",
-                "groups": [
-                  {
-                    "id": "msgReadGroup",
-                    "label": "Contoso Add-in",
-                    "icons": [
-                      { "size": 16, "url": "https://localhost:3000/assets/icon-16.png" },
-                      { "size": 32, "url": "https://localhost:3000/assets/icon-32.png" },
-                      { "size": 80, "url": "https://localhost:3000/assets/icon-80.png" }
-                    ],
-                    "controls": [
-                      {
-                        "id": "msgReadOpenPaneButton",
-                        "type": "button",
-                        "label": "Show Task Pane",
-                        "icons": [
-                          { "size": 16, "url": "https://localhost:3000/assets/icon-16.png" },
-                          { "size": 32, "url": "https://localhost:3000/assets/icon-32.png" },
-                          { "size": 80, "url": "https://localhost:3000/assets/icon-80.png" }
-                        ],
-                        "supertip": {
-                          "title": "Show Task Pane",
-                          "description": "Opens a pane displaying all available properties."
+          "id": "TaskPaneRuntimeMail",
+          "type": "general",
+          "code": {
+            "page": "https://localhost:3000/taskpane.html"
+          },
+          "lifetime": "short",
+          "actions": [
+            {
+              "id": "TaskPaneRuntimeShowMail",
+              "type":"openPage",
+              "pinnable": false,
+              "view": "dashboard"
+            }
```

The parent "runtime" object is only used in Outlook and Outlook doesn't support "view".

```suggestion
            {
              "id": "TaskPaneRuntimeShowMail",
              "type":"openPage",
              "pinnable": false
            }
```